src/Generator/IdGenerator.php

```diff
@@ -40,6 +40,6 @@ public function generatePath(MediaInterface $media): string
         $rep_first_level = (int) ($mediaId / $this->firstLevel);
         $rep_second_level = (int) (($mediaId - ($rep_first_level * $this->firstLevel)) / $this->secondLevel);
 
-        return sprintf('%s/%04s/%02s', $media->getContext(), $rep_first_level + 1, $rep_second_level + 1);
+        return sprintf('%s/%04s/%02s', $media->getContext() ?? '', $rep_first_level + 1, $rep_second_level + 1);
```

Comment:
For other cases like throwing exceptions or logging things, I don't think it matters that much if we do the ?? '' trick. But for this cases we might do an if before and throw an exception if it is null?

cc @VincentLanglet @phansys

Reply:
I don't know what's this generatePath method was used for.

For a media without context, it was casted to `''` previously. If this can't work like this it's better indeed to add an exception "Cannot generate path for media without context".

Reply:
Lets start this review again. I will resolve all your conversations and create a new ones, so we don't miss anything.